### PR TITLE
[#2073] fix(server): throws EventRetryException If Exception is encountered during create write handler

### PR DESCRIPTION
**Pull request by xianjingfeng**
### What changes were proposed in this pull request?

Throws EventRetryException If Exception is encountered during create write handler.

### Why are the changes needed?

Fix: #2073

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

CI

### CODE REVIEW

File: server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java [hunk 1]
```diff
@@ -169,7 +170,14 @@ public void processFlushEvent(ShuffleDataFlushEvent event) throws Exception {
               storageDataReplica,
               user,
               maxConcurrencyPerPartitionToWrite);
-      ShuffleWriteHandler handler = storage.getOrCreateWriteHandler(request);
+      ShuffleWriteHandler handler;
+      try {
+        handler = storage.getOrCreateWriteHandler(request);
```
Comment (zuston) [inline server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java:175]: If there is no write permission, the retry will overcome this case? Or just retry again to write this disk

Comment (zuston) [reply server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java:175]: BTW, I hope this could be tested in the UTs

Comment (xianjingfeng) [reply server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java:175]: > If there is no write permission, the retry will overcome this case? Or just retry again to write this disk

It dependency the logic of `StorageManagerSelector`. If EventRetryException be catch in `org.apache.uniffle.server.DefaultFlushEventHandler#handleEventAndUpdateMetrics`, the retryTimes of the flush event will be increase

Comment (zuston) [reply server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java:175]: Got it. +1 for this

**Review by zuston — DISMISSED** [review]

**Comment by xianjingfeng** [comment]
Should we change the default value of `rss.server.hybrid.storage.fallback.strategy.class` to `RotateStorageManagerFallbackStrategy`? @zuston @jerqi @rickyma @smallzhongfeng 
https://github.com/apache/incubator-uniffle/blob/57f0f8bbcd530feb6ab1284c2eed0c08b9e2c9ae/server/src/main/java/org/apache/uniffle/server/storage/HybridStorageManager.java#L85-L88

**Comment by xianjingfeng** [comment]
@zuston PTAL

**Comment by jerqi** [comment]
> Should we change the default value of `rss.server.hybrid.storage.fallback.strategy.class` to `RotateStorageManagerFallbackStrategy`? @zuston @jerqi @rickyma @smallzhongfeng
> 
> https://github.com/apache/incubator-uniffle/blob/57f0f8bbcd530feb6ab1284c2eed0c08b9e2c9ae/server/src/main/java/org/apache/uniffle/server/storage/HybridStorageManager.java#L85-L88

I prefer not.

**Review by jerqi — APPROVED** [review]
LGTM.

**Review by zuston — APPROVED** [review]

**Pull request merged**